contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -128,95 +128,146 @@ def _transcribe_property(prop):
         return str(prop)
 
 
-def _find_timecode_mobs(item):
-    mobs = [item.mob]
-
-    for c in item.walk():
-        if isinstance(c, aaf2.components.SourceClip):
-            mob = c.mob
-            if mob:
-                mobs.append(mob)
-            else:
-                continue
+def _find_source_clip(item, op_group_found=None):
+    if isinstance(item, aaf2.components.SourceClip):
+        return item, op_group_found
+    elif isinstance(item, aaf2.components.OperationGroup):
+        return _find_source_clip(item.segments[0], item)
+    elif isinstance(item, aaf2.components.Sequence):
+        comp = [c for c in item.components if not isinstance(c, aaf2.components.Filler)]
+        comp = comp[0] if len(comp) > 0 else None
+        return _find_source_clip(comp, op_group_found)
+    elif isinstance(item, aaf2.components.EssenceGroup):
+        choice = item['Choices'][0] if len(item['Choices']) > 0 else None
+        return _find_source_clip(choice, op_group_found)
+    elif isinstance(item, aaf2.components.NestedScope):
+        slot = item.slots[0] if len(item.slots) > 0 else None
+        return _find_source_clip(slot, op_group_found)
+    elif isinstance(item, aaf2.components.Selector):
+        alt = [a for a in item.alternates.value]
+        alt = alt[0] if len(alt) > 0 else None
+        if alt is not None:
+            return _find_source_clip(item.alternates.value[0], op_group_found)
+        # return _find_source_clip(item.selected.value, op_group_found)
```

Comment:
If alt was None, this should return something. Also, should the Selector's "Selected" item be used here instead of the first alternate?